src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1702,7 +1708,7 @@ private LLVMValueRef GetFieldAddress(FieldDesc field, bool isStatic)
 
         private void ImportLoadField(int token, bool isStatic)
         {
-            FieldDesc field = (FieldDesc)_methodIL.GetObject(token);
+            FieldDesc field = ((FieldDesc)_methodIL.GetObject(token)).GetTypicalFieldDefinition();
```

Comment:
Codegen should always look at fully instantiated things. It should never see the signature variables. Trying to load the value of field `SomeGenericType<T>.SomeField` should produce InvalidProgramException. Only `SomeGenericType<SomeArgument>.SomeField` exists. Could this be the source of trouble?

Reply:
(The `GetTypicalFieldDefinition` doesn't look right.)